GenericsHomework/GenericsHomework/Node.cs

```diff
@@ -0,0 +1,59 @@
+﻿namespace GenericsHomework;
+public class Node<T>
+{
+    private int Size { get; set; }
+    public T? Item { get; set; }
+    public Node<T> Next
+    {get; private set; }
+
+    public Node(T? item, Node<T> next)
```

Comment:
```suggestion
    public Node(T item)
```
consider not allowing null items